### ISSUE TITLE
standardbounties: where do we store info that's not in the bounty detail spec?

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#253]
should we be storing information that's missing from the database (like the githubusername, expereience level required... etc) in this JSON blob? its got to go somewhere..

 https://github.com/gitcoinco/web/pull/140#discussion_r160816196

**Comment by jasonrhaas** [comment]
That data is all being stored in the database.  We can store it in IPFS as well but I'm always hesitant to store data redundantly in two places, it will inevitably get out of sync at some point, and then need to figure out which dataset to use.  I mentioned this in another issue but at some point a decision needs to be made whether we want to use IPFS/blockchain or Postgres as our source of truth.  In the meantime, most of the data is stored in Postgres, the blockchain is just there to handle the transactions really.

**Comment by jasonrhaas** [comment]
Another solution is to store the "truth" data all in IPFS, and the database is dumb to everything.  Then the application would need to pull down the JSON blob to get the issue details for example.

**Comment by owocki** [comment]
i think that treating IPFS as the source of truth makes sense to me... and then teh database is a dumb proxy.  onboard with that approach

**Comment by jasonrhaas** [comment]
ok this one is going to be a bit more of a change.  Currently I'm setting the database tables in python to match up with what is send to web3, but not really using IPFS for much.

I think the consensys here is that we should be only writing to IPFS directly, and have the database sync off of that.

This is going to require probably a new python method or two.  I was thinking this might be the right approach:

https://github.com/ipfs/py-ipfs-api

Basically make a backend method that accepts a JSON object, it then updates the right location in IPFS and returns a hash.

Currently this is done on the front-end in Javascript.  But since we want the database tables to sync off of IPFS, I think it makes the most sense to move it into the Python code?  @mbeacom @owocki thoughts?

**Comment by jasonrhaas** [comment]
This is fixed, pending the finalization of the IPFS schema.

https://github.com/ConsenSys/StandardBounties/issues/21

**Comment by owocki** [comment]
just bumped @mbeylin on updating the docs.. not sure if that blocks this though

**Comment by jasonrhaas** [comment]
I suppose it does block this issue.  Its working with the current format that is in there, should be trivial to change to using the new schema.

**Comment by jasonrhaas** [comment]
I think this one can be closed.
